mozilla/blurts-server
# MNTOR-1800: unsubscribe from free monthly report

# References:

Jira: MNTOR-3582



# Description
1. Util function for table `subscriber_email_preferences`
The util funcs take care of the fetches and updates of email preferences. It gets the stored settings from 2 tables and consolidate them. Follow up work will be tracked in MNTOR-3557

2. Util function to generate unsubscribe URL
3. API endpoint to verify unsub url and unsub the user from monthly monitor reports


# Screenshot (if applicable)

Not applicable.

# How to test
* get the output of `console.info(generateUnsubscribeLink(session?.user.email || ""));` on any page
* visit that link
  * if the pref table is empty, a row will be created
  * if the pref table record already exists, it will check for the value of `monthly_monitor_report_free` value and change it to false if not

# Checklist (Definition of Done)
